packages/frontend/src/services/NonFungibleTokens.js

```diff
@@ -45,30 +55,54 @@ export default class NonFungibleTokens {
             });
         }
         // TODO: Separate Redux action for loading image
-        tokens = await Promise.all(tokens.filter(({ metadata }) => !!metadata).map(async ({ metadata, ...token }) => {
-            const { media } = metadata;
-            let mediaUrl;
-            if (!media.includes('://')) {
-                if (base_uri) {
-                    mediaUrl = `${base_uri}/${media}`;
-                } else {
-                    mediaUrl = media.startsWith('data:image') ? media : `https://cloudflare-ipfs.com/ipfs/${media}`;
-                }
-            } else {
-                mediaUrl = media;
-            }
-
-            return {
-                ...token,
-                metadata: {
-                    ...metadata,
-                    mediaUrl
-                }
-            };
-        }));
+        tokens = tokens
+          .filter(({ metadata }) => !!metadata)
+          .map(token => mapTokenMediaUrl(token, base_uri));
 
         return tokens;
     }
+
+    static transfer = async ({ accountId, contractId, tokenId, receiverId }) => {
+        console.log('transfer()', accountId, contractId, tokenId, receiverId);
+        const account = await wallet.getAccount(accountId);
+
+        return account.signAndSendTransaction({
```

Comment:
got an error when calling this:
```
TypeError: First argument must be a string, Buffer, ArrayBuffer, Array, or array-like object.
    at fromObject (index.js:311)
    at from (index.js:137)
    at Function.Buffer.from (index.js:149)
    at BinaryWriter.writeString (index.js:99)
    at serializeField (index.js:219)
    at index.js:265
    at Array.map (<anonymous>)
    at serializeStruct (index.js:264)
    at Object.serialize (index.js:287)
    at signTransactionObject (transaction.js:214)
```

any idea what's going wrong?

Reply:
I'm not able to reproduce this. Can you please provide some more context?

Reply:
turns out I forgot to run yarn install lol